src/operator/timeout.ts

```diff
@@ -20,7 +20,8 @@ export function timeout<T>(this: Observable<T>, due: number | Date,
                            scheduler: Scheduler = async): Observable<T> {
   const absoluteTimeout = isDate(due);
   const waitFor = absoluteTimeout ? (+due - scheduler.now()) : Math.abs(<number>due);
-  const error = errorToSend || new TimeoutError();
+  const error = errorToSend instanceof Error ? errorToSend : new TimeoutError(errorToSend);
```

Comment:
Let me correct myself on two points:  (It was late.  :)

1) Sorry, by "specialization" , what I meant was that the _usage_ of the .throw() and .timeout() operators is similar (or should be, IMO) with regard to the supplied error value.  In both cases, the caller creates and supplies the error value in the context of the operator's instantiation rather than the operation's later execution.  This is sometimes a problem (esp. when creating an ```Error``` instance is desiring an execution-relative call stack), but not necessarily so.

2) The old Rx API, at least Rx4, despite what I said, does throw an ```Error``` instance from .timeout() (with .name set to "TimeoutError").  So, Rx4, too, has an opinion about what the error value should be in this case.  Yuck.

I think the way .timeout() works is a mistake, in both the previous Rx versions and also now the "improved" version in Rx5.  Rx should have no opinion about any error value.  Rx should not require or assume an error value is an instance of ```Error``` (Javascript does not).  If I want to time-out my streams using a string, number, or other object as the error value, I should be able to (without Rx getting in my way with "helpful" wrappers or assumptions).

So...  perhaps not in this release, if we want to "fix" or improve the situation with regards to the inability to create the error value in the context of the .timeout() (or even .throw()) operator's execution, we might consider having those operators accept an "error value creation" callback function as I described earlier.

Having .throw() forward the error value supplied to it as-is and .timeout() do something entirely different with the error value supplied it, seems rather inconsistent.